From 7e9af59964242eb87b4e702dc63f78c0f203226d Mon Sep 17 00:00:00 2001
From: Ali Polatel <hawking@gentoo.org>
Date: Sun, 26 Oct 2008 16:17:21 +0200
Subject: [PATCH 05/10] Replace lib with @@EXHERBO_LIBDIR@@ for multilib.
Organization: Gentoo

Needs cleaning up. The automake parts probably can be improved.
See what debian does.

---
 Lib/distutils/command/install.py |    4 ++--
 Lib/distutils/sysconfig.py       |    3 ++-
 Lib/site.py                      |    4 ++--
 Makefile.pre.in                  |    4 ++--
 Modules/Setup.dist               |    2 +-
 Modules/getpath.c                |    6 +++---
 setup.py                         |    4 ++--
 7 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/Lib/distutils/command/install.py b/Lib/distutils/command/install.py
index 76efe3b..865af0a 100644
--- a/Lib/distutils/command/install.py
+++ b/Lib/distutils/command/install.py
@@ -41,8 +41,8 @@ else:
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
-        'platlib': '$platbase/lib/python$py_version_short/site-packages',
+        'purelib': '$base/@@EXHERBO_LIBDIR@@/python$py_version_short/site-packages',
+        'platlib': '$platbase/@@EXHERBO_LIBDIR@@/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index 2db4ab5..aa4e37d 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -116,7 +116,8 @@ def get_python_lib(plat_specific=0, standard_lib=0, prefix=None):
 
     if os.name == "posix":
         libpython = os.path.join(prefix,
-                                 "lib", "python" + get_python_version())
+                                "@@EXHERBO_LIBDIR@@",
+                                "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
diff --git a/Lib/site.py b/Lib/site.py
index 21c7db2..2a4bf28 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -265,10 +265,10 @@ def addsitepackages(known_paths):
         if sys.platform in ('os2emx', 'riscos'):
             sitedirs.append(os.path.join(prefix, "Lib", "site-packages"))
         elif os.sep == '/':
-            sitedirs.append(os.path.join(prefix, "lib",
+            sitedirs.append(os.path.join(prefix, "@@EXHERBO_LIBDIR@@",
                                         "python" + sys.version[:3],
                                         "site-packages"))
-            sitedirs.append(os.path.join(prefix, "lib", "site-python"))
+            sitedirs.append(os.path.join(prefix, "@@EXHERBO_LIBDIR@@", "site-python"))
         else:
             sitedirs.append(prefix)
             sitedirs.append(os.path.join(prefix, "lib", "site-packages"))
diff --git a/Makefile.pre.in b/Makefile.pre.in
index 8451445..d06be9d 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -87,11 +87,11 @@ datarootdir=    @datarootdir@
 
 # Expanded directories
 BINDIR=		$(exec_prefix)/bin
-LIBDIR=		$(exec_prefix)/lib
+LIBDIR=		$(exec_prefix)/@@EXHERBO_LIBDIR@@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
-SCRIPTDIR=	$(prefix)/lib
+SCRIPTDIR=	$(prefix)/@@EXHERBO_LIBDIR@@
 
 # Detailed destination directories
 BINLIBDEST=	$(LIBDIR)/python$(VERSION)
diff --git a/Modules/Setup.dist b/Modules/Setup.dist
index c67dad4..cf35a7a 100644
--- a/Modules/Setup.dist
+++ b/Modules/Setup.dist
@@ -455,7 +455,7 @@ GLHACK=-Dclear=__GLclear
 # Andrew Kuchling's zlib module.
 # This require zlib 1.1.3 (or later).
 # See http://www.gzip.org/zlib/
-#zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+#zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/@@EXHERBO_LIBDIR@@ -lz
 
 # Interface to the Expat XML parser
 #
diff --git a/Modules/getpath.c b/Modules/getpath.c
index 09fbe10..234f2f4 100644
--- a/Modules/getpath.c
+++ b/Modules/getpath.c
@@ -129,7 +129,7 @@ static char prefix[MAXPATHLEN+1];
 static char exec_prefix[MAXPATHLEN+1];
 static char progpath[MAXPATHLEN+1];
 static char *module_search_path = NULL;
-static char lib_python[] = "lib/python" VERSION;
+static char lib_python[] = "@@EXHERBO_LIBDIR@@/python" VERSION;
 
 static void
 reduce(char *dir)
@@ -524,7 +524,7 @@ calculate_path(void)
     }
     else
         strncpy(zip_path, PREFIX, MAXPATHLEN);
-    joinpath(zip_path, "lib/python00.zip");
+    joinpath(zip_path, "@@EXHERBO_LIBDIR@@/python00.zip");
     bufsz = strlen(zip_path);	/* Replace "00" with version */
     zip_path[bufsz - 6] = VERSION[0];
     zip_path[bufsz - 5] = VERSION[2];
@@ -534,7 +534,7 @@ calculate_path(void)
             fprintf(stderr,
                 "Could not find platform dependent libraries <exec_prefix>\n");
         strncpy(exec_prefix, EXEC_PREFIX, MAXPATHLEN);
-        joinpath(exec_prefix, "lib/lib-dynload");
+        joinpath(exec_prefix, "@@EXHERBO_LIBDIR@@/lib-dynload");
     }
     /* If we found EXEC_PREFIX do *not* reduce it!  (Yet.) */
 
diff --git a/setup.py b/setup.py
index 951bb50..81791de 100644
--- a/setup.py
+++ b/setup.py
@@ -594,11 +594,11 @@ class PyBuildExt(build_ext):
             elif self.compiler.find_library_file(lib_dirs, 'curses'):
                 readline_libs.append('curses')
             elif self.compiler.find_library_file(lib_dirs +
-                                               ['/usr/lib/termcap'],
+                                               ['/usr/@@EXHERBO_LIBDIR@@/termcap'],
                                                'termcap'):
                 readline_libs.append('termcap')
             exts.append( Extension('readline', ['readline.c'],
-                                   library_dirs=['/usr/lib/termcap'],
+                                   library_dirs=['/usr/@@EXHERBO_LIBDIR@@/termcap'],
                                    extra_link_args=readline_extra_link_args,
                                    libraries=readline_libs) )
         else:
-- 
1.6.0.2

